contributing/contribution_guide.rst

```diff
@@ -0,0 +1,393 @@
+relevant items:
+
+.. code-block:: text
+
+    - [ ] Fix the specs as they have not been updated yet
+    - [ ] Submit changes to the documentation
+    - [ ] Document the BC breaks
+
+If the code is not finished yet because you don't have time to finish it or
+because you want early feedback on your work, add an item to todo-list:
+
+.. code-block:: text
+
+    - [ ] Finish the feature
+    - [ ] Gather feedback for my changes
+
+As long as you have items in the todo-list, please prefix the pull request
+title with "[WIP]".
+
+In the pull request description, give as much details as possible about your
+changes (don't hesitate to give code examples to illustrate your points). If
+your pull request is about adding a new feature or modifying an existing one,
+explain the rationale for the changes. The pull request description helps the
+code review.
+
+In addition to this "code" pull request, you must also send a pull request to
+the `documentation repository`_ to update the documentation when appropriate.
+
+Rework your Patch
+~~~~~~~~~~~~~~~~~
+
+Based on the feedback on the pull request, you might need to rework your
+patch. Before re-submitting the patch, rebase with ``upstream/master``, don't merge; and force the push to the origin:
+
+.. code-block:: bash
+
+    $ git rebase -f upstream/master
+    $ git push --force origin BRANCH_NAME
+
+.. note::
+
+    When doing a ``push --force``, always specify the branch name explicitly
+    to avoid messing other branches in the repo (``--force`` tells Git that
+    you really want to mess with things so do it carefully).
+
+Often, Akeneo team members will ask you to "squash" your commits. This means you will
+convert many commits to one commit. To do this, use the rebase command:
+
+.. code-block:: bash
+
+    $ git rebase -i upstream/master
+    $ git push --force origin BRANCH_NAME
+
+After you type this command, an editor will popup showing a list of commits:
+
+.. code-block:: text
+
+    pick 1a31be6 first commit
+    pick 7fc64b4 second commit
+    pick 7d33018 third commit
+
+To squash all commits into the first one, remove the word ``pick`` before the
+second and the last commits, and replace it by the word ``squash`` or just
+``s``. When you save, Git will start rebasing, and if successful, will ask
+you to edit the commit message, which by default is a listing of the commit
+messages of all the commits. When you are finished, execute the push command.
+
+.. _`Akeneo PIM repository`:            https://github.com/akeneo/pim-community-dev
+.. _ProGit:                             http://git-scm.com/book
+.. _GitHub:                             https://github.com/signup/free
+.. _`GitHub's Documentation`:           https://help.github.com/articles/ignoring-files
+.. _`documentation repository`:         https://github.com/akeneo/pim-docs
+.. _`Sylius documentation`:             http://docs.sylius.org/en/latest/contributing/index.html
```

Comment:
May be we can delete "Largely inspired note" from the beginning if we have one here